internal/command/meta_backend_migrate.go

```diff
@@ -569,10 +569,20 @@ func (m *Meta) backendMigrateTFC(opts *backendMigrateOpts) error {
 		opts.sourceWorkspace = currentWorkspace
 
 		log.Printf("[INFO] backendMigrateTFC: single-to-single migration from source %s to destination %q", opts.sourceWorkspace, opts.destinationWorkspace)
-		// Run normal single-to-single state migration
+		// Run normal single-to-single state migration.
 		// This will handle both situations where the new cloud backend
 		// configuration is using a workspace.name strategy or workspace.tags
 		// strategy.
+		//
+		// We do prompt first though, because state migration is mandatory
+		// for moving to Cloud and the user should get an opportunity to
+		// confirm that first.
+		if migrate, err := m.promptSingleToCloudSingleStateMigration(opts); err != nil {
+			return err
+		} else if !migrate {
+			return nil //skip migrating but return successfully
+		}
```

Comment:
Honestly, I'm not totally sure... when I first tested this I found some paths that were previously not prompting at all, and that's why I added this here. It looks like the other function belongs to a different migration function that handles some different cases, but I agree we can probably consolidate these two functions together if their error messages are equivalent enough.

Can we hold this for a later change, since it seems like it'll just be a pretty trivial in-place swap to the other function and then delete the redundant one I added here?

Reply:
Follow-up: I added an extra commit to avoid printing out the redundant migration message this implies in some paths, so I'm hoping that's a reasonable _immediate_ solution which we can polish in a later commit.